Client/Frontend/Toolbar+URLBar/TabLocationView.swift

```diff
@@ -435,6 +435,8 @@ extension TabLocationView: TabEventHandler {
             trackingProtectionButton.alpha = 1.0
             let themeManager: ThemeManager = AppContainer.shared.resolve()
             self.blockerStatus = blocker.status
+            // We need to hide the trackingProtectionButton when the user activates the reader mode(Bug 16400). When user turns on the reader mode the url is changed to localhost hence we check whether url contains http://localhost or not, according to this we will hide or show the trackingProtectionButton.
+            self.trackingProtectionButton.isHidden = tab.webView?.url?.absoluteString.contains("http://localhost") ?? false
```

Comment:
Instead of checking for the url we should check against `readerModeState` instead. Also this code should go into `setReaderModeState(:)` as `updateBlockerStatus(tab:)` is not always called when reader mode is activated.

Reply:
Hey @thatswinnie. Thanks a lot. I will fix this and push the code again